spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright 2017-2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.core.subscriber;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.cloud.gcp.pubsub.support.AcknowledgeablePubsubMessage;
+import org.springframework.cloud.gcp.pubsub.support.PubSubAcknowledger;
+import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
+import org.springframework.util.Assert;
+
+/**
+ * Default implementation of {@link PubSubSubscriberOperations}.
+ *
+ * <p>The main Google Cloud Pub/Sub integration component for consuming
+ * messages from subscriptions asynchronously or by pulling.
+ *
+ * @author Vinicius Carvalho
+ * @author João André Martins
+ * @author Mike Eltsufin
+ * @author Chengyuan Zhao
+ * @author Doug Hoard
+ * @since 1.1.0
+ */
+public class PubSubSubscriberTemplate implements PubSubSubscriberOperations, InitializingBean {
+
+	private static final Log LOGGER = LogFactory.getLog(PubSubSubscriberTemplate.class);
```

Comment:
Not used?

Reply:
Removed instance of LOGGER since it is not used.